framework/auth/core.py

```diff
@@ -633,32 +683,50 @@ def is_active(self):
                 self.is_confirmed)
 
     def get_unclaimed_record(self, project_id):
-        """Get an unclaimed record for a given project_id.
+        """
+        Get an unclaimed record for a given project_id.
 
         :raises: ValueError if there is no record for the given project.
         """
         try:
             return self.unclaimed_records[project_id]
-        except KeyError:  # reraise as ValueError
+        except KeyError:  # re-raise as ValueError
             raise ValueError('No unclaimed record for user {self._id} on node {project_id}'
-                                .format(**locals()))
+                             .format(**locals()))
+
+    def verify_claim_token(self, token, project_id):
+        """
+        Verify the claim token for this user for a given node
+        which she/he was added as a unregistered contributor for.
+
+        :param token: the verification token
+        :param project_id: the project node id
+        :rtype boolean
+        :return: whether or not a claim token is valid
+        """
+        try:
+            record = self.get_unclaimed_record(project_id)
+        except ValueError:  # No unclaimed record for given pid
+            return False
+        return record['token'] == token and record['expires'] > dt.datetime.utcnow()
 
     def get_claim_url(self, project_id, external=False):
-        """Return the URL that an unclaimed user should use to claim their
-        account. Return ``None`` if there is no unclaimed_record for the given
-        project ID.
-
-        :param project_id: The project ID for the unclaimed record
-        :raises: ValueError if a record doesn't exist for the given project ID
-        :rtype: dict
-        :returns: The unclaimed record for the project
         """
+        Return the URL that an unclaimed user should use to claim their account.
+        Raise `ValueError` if there is no unclaimed_record for the given project ID.
+
+        :param project_id: the project ID for the unclaimed record
+        :param external:
+        :rtype: string
+        :returns: the claim url
+        :raises: ValueError if there is no record for the given project.
```

Comment:
For CR: not sure why explicitly throw `ValueError` instead of `return None`, and there is a also unit test for this